fiddler/datadog_checks/fiddler/check.py

```diff
@@ -0,0 +1,164 @@
+import time
+
+import fiddler as fdl
+
+from datadog_checks.base import AgentCheck
+
+# from typing import Any
+
+
+# We will get one hours worth data in chunks and process them.
+bin_size = 3600
+
+metrics_list = ['accuracy']
+
+
+def create_tags(**tags_in):
+    tags = []
+
+    tags.append("project:" + tags_in['project'])
+    tags.append("model:" + tags_in['model'])
+    if len(tags_in) > 2:
+        tags.append("feature:" + tags_in['feature'])
+
+    return tags
+
+
+class FiddlerCheck(AgentCheck):
+
+    # This will be the prefix of every metric and service check the integration sends
+    __NAMESPACE__ = 'fiddler'
+
+    def __init__(self, name, init_config, instances):
+        super(FiddlerCheck, self).__init__(name, init_config, instances)
+
+        # Connection parameters for Fiddler. Will be parameterizing this later.
+        self.URL = 'https://demo.fiddler.ai'
+        self.ORG_ID = 'demo'
+        self.AUTH_TOKEN = 'K4ph7ORDcIO2xVIEA6KxL1o1zHjZockgurhCOZOUSVs'
+
+        self.log.info("Inside init method.")
+        #        self.client = fdl.FiddlerApi(url=self.URL, org_id=self.ORG_ID, auth_token=self.AUTH_TOKEN)
+
+        self.base_url = self.instance.get('url')
+        self.api_key = self.instance.get('fiddler_api_key')
+        self.org = self.instance.get('organization')
+
+        #            self.log.info("Connecting to : ", self.base_url)
+        #            self.log.info("with org id : ", self.org)
+        #            self.log.info("and auth key : ", self.api_key)
+
+        print("Connecting to : ", self.base_url)
+        print("with org id : ", self.org)
+        print("and auth key : ", self.api_key)
+
+        self.client = fdl.FiddlerApi(url=self.base_url, org_id=self.org, auth_token=self.api_key)
+
+    # Use self.instance to read the check configuration
+
+    def check(self, _):
+        # Iterate through the projects and the models and push data into Fiddler
+        project_path = ['list_projects', self.ORG_ID]
+        result_all = self.client.v1._call(project_path)
+        #        self.log.info("Projects: ", result_all["projects"])
+
+        start_time = (time.time() * 1000) - (bin_size * 1000)
+        end_time = time.time() * 1000
+        self.log.info("Start time is : ", start_time)
+        self.log.info("End time is : ", end_time)
+
+        # Iterate through all of the projects within the Fiddler instance and get the metrics
+        for project in result_all["projects"]:
+            models = project["models"]
+
+            # Iterate through all of the models within a project
+            for model in models:
+                self.log.info("Model: ", model["id"])
+                for metric in metrics_list:
```

Comment:
Looking at https://github.com/DataDog/integrations-extras/blob/23a9997147935cde27a0e1ac80edda7026c1f019/fiddler/datadog_checks/fiddler/check.py#L13

`metrics_list` only has one element, but looking at this loop and the `if-elseif` chain below, it looks as if it should have more. As it is, I think that a lot of code below will never be reached because `metric` will only ever take the value `accuracy`.

Reply:
@alopezz thanks for all of the comments. Yes, I have only added one metrics_list in my list; because I want to get one successful test where I can see the accuracy metrics get pushed to a datadog instance. All integrated with a working datadog agent after installing the fiddler integration. However that has not happened and hence I have not added the other metrics. I will add them and finish of the testing once I have gotten over one round of testing. 

I installed the integration on a new EC2 instance to do some testing. And here are the results after I have installed the fiddler-client using the datadog-agent's pip. As you can see below, I can import the fiddler from there. 

[ec2-user@ip-172-31-90-254 fiddler.d]$ pwd
/etc/datadog-agent/conf.d/fiddler.d
[ec2-user@ip-172-31-90-254 fiddler.d]$ cd /opt/datadog-agent/embedded/bin
[ec2-user@ip-172-31-90-254 bin]$ ./pip freeze | grep fiddler
datadog-fiddler @ file:///tmp/datadog_fiddler-1.0.0-py2.py3-none-any.whl
fiddler-client==1.4.0
[ec2-user@ip-172-31-90-254 bin]$ ./python -c "import fiddler"
[ec2-user@ip-172-31-90-254 bin]$ 

But even with the above, when I start the datadog-agent, I can see the below error in the log file. 

2022-09-26 15:55:16 UTC | CORE | ERROR | (pkg/collector/scheduler.go:201 in getChecks) | Unable to load a check from instance of config 'fiddler': JMX Check Loader: check is not a jmx check, or unable to determine if it's so; Python Check Loader: unable to import module 'fiddler': No module named 'fiddler'; Core Check Loader: Check fiddler not found in Catalog
2022-09-26 15:55:16 UTC | CORE | ERROR | (pkg/collector/scheduler.go:248 in GetChecksFromConfigs) | Unable to load the check: unable to load any check from config 'fiddler'
2022-09-26 15:55:16 UTC | CORE | INFO | (pkg/collector/scheduler/scheduler.go:92 in Enter) | Scheduling check cpu with an interval of 15s

Let me know what I should be trying out. Thanks.

Reply:
Hey @rajeevthefiddler, there are bugs in the `check.py` code that is causing the agent to not be able to compile the python. To debug, please run the check manually with debug level logging so you can get more information about why the check isn't running: `sudo -u dd-agent datadog-agent check fiddler -l debug`. 

When I run the check in the fiddler E2E agent, I see this error in the logs: 

```
Traceback (most recent call last):
  File "/home/fiddler/datadog_checks/fiddler/check.py", line 36, in __init__
    self.log.info("Connecting to : ", self.base_url)
[...]
TypeError: not all arguments converted during string formatting
```
Which is pointing to the string formatting issue here: https://github.com/DataDog/integrations-extras/pull/1383/files#r980300745.

You'll need to fix the bugs that prevent the agent from compiling the python before the agent can run the check so please use above mentioned manual check run command to debug the code.